### PR TITLE
feat:Activity-Hydrogen Production

### PR DESCRIPTION
**Pull request by ayeshmcg**
Card: https://github.com/bcgov/cas-reporting/issues/289. This adds the Hydrogen production activity.

**Changes:-**
- Adds backend config/migration for Hydrogen production activity.
- Adds frontend page and components for Hydrogen production activity.
- Added custom_methodology_schema model for handling complex methodology logic.
- Updated build_form_schema to apply custom schemas dynamically.
- Added any missing configurations and relationships to the database.Created routing for industry_user and industry_user_admin flows.
- Added components for activity-specific pages and dynamic field rendering.

**To test:-**
- make reset_db to apply latest migrations.
- spin up the front, backend and postgres.
- Login, then navigate to http://localhost:3000/reporting/reports/1/facilities/1/activities/hydrogemr_production, click through the available form fields.
- The Gas type dropdown options should change options available in the methodology field.
- The Methodology dropdown options spawn additional fields, depending on the option selected.

### CODE REVIEW

**Review by dleard — CHANGES_REQUESTED** [review]
Nice! It all works correctly, just a few changes needed & I've tagged a couple other devs in places where I have questions for them to weigh in on

File: bciers/apps/reporting/src/app/components/activities/hydrogenProduction.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  reportDate: string;
+}
+
+// 🧩 Main component
+export default function HydrogenProduction({
+  activityData,
+  reportDate,
+}: Readonly<Props>) {
+  const defaultEmptySourceTypeState = {
+    emissions: [{ gasType: "CO2" }],
```
[the remark this answers is not in the document]
Comment (ayeshmcg) [reply bciers/apps/reporting/src/app/components/activities/hydrogenProduction.tsx]: @dleard  Removed this for now

File: bc_obps/reporting/migrations/0016_custom_methodology_schema.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Generated by Django 5.0.7 on 2024-09-12 23:15
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('registration', '0040_alter_facility_address_and_more'),
```
[the remark this answers is not in the document]
Comment (ayeshmcg) [reply bc_obps/reporting/migrations/0016_custom_methodology_schema.py:10]: @dleard shall I keep this for now?

[the remark this answers is not in the document]
Comment (Sepehr-Sobhani) [reply bc_obps/reporting/migrations/0016_custom_methodology_schema.py:10]: If Django added the dependency automatically, it's best to keep it. Django uses the dependency array to build a dependency graph, so it's usually best not to manually update migration files unless you really need to.

File: bc_obps/reporting/tests/models/program_configuration_tests/test_hydrogen_production_2024.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                    actual_methodologies,
+                    expected_methodologies,
+                    ordered=False,
+                    msg=f'{source_type_name}:{gas_name} contains config for the proper methods and field counts',
+                )
+
+        # Ensure there are exactly 4 configuration elements
+        self.assertEqual(len(config_elements), 4)
+
+    def testCustomMethodologySchema(self):
```
Comment (dleard) [inline bc_obps/reporting/tests/models/program_configuration_tests/test_hydrogen_production_2024.py:84]: :heart:

**Review by dleard — APPROVED** [review]
Looks great! Thanks for all your work on the custom methodology schema stuff, that unblocks a couple other activities now that will also need it :)

**Pull request merged**